jenkins_pipelines/environments/common/pipeline-build-validation.groovy

```diff
@@ -272,8 +273,16 @@ def clientTestingStages() {
                 echo "Testing ${minion}"
             }
             if (params.must_add_MU_repositories) {
-                stage('Add MUs') {
-                    if (!minion.contains('ssh')) {
+                stage("Add_MUs_${minion}") {
+                    if (minion.contains('ssh_minion')) {
+                        // SSH minion need minion MU channel. This section wait until minion finish creating MU channel
+                        def minion_name_without_ssh = minion.replaceAll('ssh_minion', 'minion')
+                        println "Waiting for the MU channel creation by ${minion_name_without_ssh} for ${minion}."
+                        waitUntil {
+                            mu_sync_status[minion_name_without_ssh]
+                        }
+                        println "MU channel available for ${minion} "
```

Comment:
It might be good to extract all this as a method.
For instance:
`waitUntilRepoIsReady(minion)`

Reply:
I'm not sure it will be possible to create a function. The `mu_sync_status` variable is define for the function. If this variable is updated during clientTestingStages the waitUntilRepoIsReady method will probably not see it.